langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -868,6 +868,45 @@ function testFromJsonWithTypeTable() {
 
 }
 
+type IntVal record { int? x;};
+
+type Cases record {
+    Case[] Data;
+    string LastData;
+    string Source;
+    string DevBy;
+};
+
+type Case record {
+  string ConfirmDate;
+  string No;
+  int? Age;
+  string Nation;
+  string? Detail;
+};
+
+function testFromJsonWithTypeWithNullValues() {
+    json j1 = {x: null};
+    IntVal val = checkpanic j1.fromJsonWithType(IntVal);
+    assert(val.toString(), "{\"x\":null}");
+
+    json j2 =  {"Data":
```

Comment:
This PR fixes https://github.com/ballerina-platform/ballerina-lang/issues/26117 too, right? Shall we use the JSON value from that instead? We can maybe change a field in `Address` to be nilable and change the JSON value a bit too.

Please mention the issue in the PR too.